libwasmvm/src/cache.rs

```diff
@@ -384,6 +385,64 @@ fn do_get_metrics(cache: &mut Cache<GoApi, GoStorage, GoQuerier>) -> Result<Metr
     Ok(cache.metrics().into())
 }
 
+#[derive(Serialize)]
+struct PerModuleMetrics {
+    hits: u32,
+    size: usize,
+}
+
+impl From<cosmwasm_vm::PerModuleMetrics> for PerModuleMetrics {
+    fn from(value: cosmwasm_vm::PerModuleMetrics) -> Self {
+        Self {
+            hits: value.hits,
+            size: value.size,
+        }
+    }
+}
+
+#[derive(Serialize)]
+struct PinnedMetrics {
+    per_module: HashMap<Checksum, PerModuleMetrics>,
```

Comment:
If we go back to a map here, I think it would be more elegant to not have the vector in `cosmwasm_vm::PinnedMetrics` but keep the map as well

Reply:
I was just turning it into a map to try out different things when Go refused to deserialize the MessagePack correctly. The map construction can *probably* be skipped. I'll have a look.

Reply:
Removed the intermediate `HashMap` representation